olive/cache.py

```diff
@@ -9,82 +9,103 @@
 from pathlib import Path
 from typing import Dict, Optional, Union
 
-from olive.common.config_utils import serialize_to_json
-from olive.common.utils import hash_dict
-from olive.resource_path import ResourcePath, create_resource_path
+from olive.common.config_utils import ConfigBase, convert_configs_to_dicts, serialize_to_json, validate_config
+from olive.common.utils import hash_dict, set_nested_dict_value
+from olive.resource_path import ResourcePath, create_resource_path, find_all_resources
 
 logger = logging.getLogger(__name__)
 
 
-def get_cache_sub_dirs(cache_dir: Union[str, Path] = ".olive-cache"):
-    """Return the subdirectories of the cache directory.
+def set_cache_dir(cache_dir: Union[str, Path]):
```

Comment:
open to discuss: it is a good idea to set the cache folder with os.env. Also, think further, a cli interface to clean cache folder can be useful, just like huggingface did.

Reply:
We could add that interface. But I think that could be a future work item?

Reply:
Yeah, just an idea.